src/modules/dpos/validator/components/EditVote/index.js

```diff
@@ -1,148 +1,235 @@
-  const existingVote = useSelector((state) => state.voting[address || wallet.summary.address]);
-  const [voteAmount, setVoteAmount] = useVoteAmountField(
-    existingVote ? fromRawLsk(existingVote.unconfirmed) : '',
-  );
-  const mode = existingVote ? 'edit' : 'add';
+const EditVote = ({ history, voteEdited, network, voting }) => {
+  const { t } = useTranslation();
+  const [
+    {
+      metadata: { address: currentAddress },
+    },
+  ] = useCurrentAccount();
   const [maxAmount, setMaxAmount] = useState(0);
+  const [isForm, setIsForm] = useState(true);
+
+  const [address] = selectSearchParamValue(history.location.search, ['address']);
+  const delegateAddress = address || currentAddress;
+
+  const { data: delegates, isLoading: isLoadingDelegates } = useDelegates({
+    config: { params: { address: delegateAddress } },
+  });
+
+  const delegate = useMemo(() => delegates?.data?.[0] || {}, [isLoadingDelegates]);
+  const delegatePomHeight = useMemo(() => delegate.pomHeights?.[0] || {}, [delegate]);
+
+  const { data: blocks } = useBlocks({ config: { params: { limit: 1 } } });
+  const currentHeight = useMemo(() => blocks?.data?.[0]?.height, [blocks]);
+
+  const { data: sentVotes } = useSentVotes({
+    config: { params: { address: currentAddress } },
+  });
+
+  const { data: tokens } = useTokensBalance({ config: { params: { tokenID: dposTokenId } } });
+  const token = useMemo(() => tokens?.data?.[0] || {}, [tokens]);
+
+  const { data: authData } = useAuth({ config: { params: { address: delegateAddress } } });
+  const auth = useMemo(() => ({ ...authData?.data, ...authData?.meta }), [authData]);
+  const { nonce, publicKey, numberOfSignatures, optionalKeys = [], mandatoryKeys = [] } = auth;
+
+  const [start = delegatePomHeight.start, end = delegatePomHeight.end] = selectSearchParamValue(
+    history.location.search,
+    ['start', 'end']
+  );
+
+  const hasSentVoteToDelegate = useMemo(() => {
+    const votes = sentVotes?.data?.votes;
+    if (!votes) return false;
+
+    return votes.some(({ delegateAddress: dAddress }) => dAddress === delegateAddress);
+  }, [sentVotes]);
+
+  const [voteAmount, setVoteAmount] = useVoteAmountField(0);
+  const mode = hasSentVoteToDelegate ? 'edit' : 'add';
+  const titles = getTitles(t)[mode];
+
   useEffect(() => {
-    getMaxAmount(wallet, network, voting, address || wallet.summary.address).then(
-      setMaxAmount,
-    );
-  }, [wallet, voting]);
+    getMaxAmount({
+      balance: token.availableBalance,
+      nonce,
+      publicKey,
+      voting,
+      address,
+      network,
+      numberOfSignatures,
+      mandatoryKeys,
+      optionalKeys,
+    }).then(setMaxAmount);
+  }, [token, auth, network, voting]);
 
-  const confirm = () => {
+  const handleConfirm = () => {
+    if (!isForm) {
+      removeThenAppendSearchParamsToUrl(history, { modal: 'votingQueue' }, ['modal']);
+      return;
+    }
     voteEdited([
       {
-        address: address || wallet.summary.address,
```

Comment:
This was used to handle other delegate profiles

Reply:
added comments for easy look through